DataFormats/Detectors/ITSMFT/common/src/TopologyDictionary.cxx

```diff
@@ -83,7 +83,7 @@ ClassImp(o2::ITSMFT::TopologyDictionary)
         in.read(reinterpret_cast<char*>(&gr.mPattern.mBitmap), sizeof(unsigned char) * (ClusterPattern::kExtendedPatternBytes));
         mVectorOfGroupIDs.push_back(gr);
         if (gr.mPattern.getUsedBytes() == 1)
-          mSmallTopologiesLUT[(gr.mPattern.getRowSpan() - 1) * 255 + (int)gr.mPattern.mBitmap[2]] = groupID;
```

Comment:
@lbariogl Sorry, I seem to have pushed a wrong button, and my previous comments got lost...

I was commenting about the lines 73 and 74 above.
1) I think we should better use "return number" instead of "exit(1)".  So that the error could be intercepted by the caller code. It is probably not good to "exit()", because we will run together with other detectors.
2) Also, when printing an error message, we should use LOG(ERROR). Not just the std::cout.

Cheers,
Yura.

Reply:
Thank you @iouribelikov for your comments. I replaced `exit(1)` with `trow excpetion` in all my classes and used `LOG(ERROR)`.

Cheers,
Luca

Reply:
@lbariogl  Ciao Luca.  Thank you very much !
But, now, don't we have to catch these exceptions everywhere in the calling code ?
For example, in the run_clus_itsSA.C macro ?   What will happen when we run for the first time,
and no topology dictionary is provided yet ?
(Doing a modest "return number" looked to me more "backward compatible".)

Cheers,
Yura.

Reply:
Hi, I would say an exception here would be fine since the method signature is not "noexcept" :) Since the presence of the file seems rather important here an exception is quite natural as it is easier to spot (and debug) than bugs introduced downstream by an uncaught (or badly handled) return code. Btw, since the method clearly causes side effects that are not clear from the signature (no output, input is a string) probably it should not be exposed as an external API and the exception should be handled internally to the class as well (even if it is done by rethrowing :) )

Reply:
@mkrzewic Hi Mikolaj,
Thank you very much for having a look at this PR !

If I understand correctly (which is not 100% guaranteed), we need to have a possibility to jump over this error, when running "for the very fist time".  Because, we need to get some clusters, before any topology dictionary becomes available.
There can probably be several possible solutions. But, likely, we cannot just unconditionally stop the cluster finder at this point.

Cheers,
Yura.

Reply:
@iouribelikov The problem appears just when we want to identify the topology and the dictionary has alredy been generated. In particular, the exception is thrown when the file containing the dictionary cannot be opened. If the dictionary is not loaded, then it is not possible to search in the dictionary and a new file must be provided. This is the reason while at first I used exit(). Here the olny solution that I can see is to provide a new file when this problem occurres, i.e. exit + reload. Do you think that other solutions are possible?

Cheers,
Luca

Reply:
@lbariogl Hi Luca,

How do we test this PR ?
When I tried the run_clus_itsSA.C macro, it did not go to the end because of the  exit(1).   
Probably, it will not work with the uncaught exception either.

What is the correct "entry point" to this PR ?

Thanks a lot and cheers,
Yura.

Reply:
@iouribelikov You are right. I need to add a flag within the macro, which is used both before and after the generation of the dictionary. I will correct it.

Cheers,
Luca

Reply:
@lbariogl  Great ! This is what "several solutions are possible" means :-)
Ciao,
Yura.

Reply:
@iouribelikov I have added a flag `withDictionary`. first time, run the macro with the flag set to _false_ (default). Then, generate the dictionary using _CheckTopologies.C_. Finally, run _run_clus_itsSA.C_ with the flag set to _true_. You can check the distribution of the topologies ith the new macro _compareTopologyDistributions.C_. Everything is reported also as comment in _run_clus_itsSA.C_.

Thank you and cheers,
Luca